client/entry.js

```diff
@@ -1,37 +1,28 @@
 import View from './view'
 import * as axios from 'axios'
 import * as Colyseus from 'colyseus.js'
-import Controller from 'key-controller'
 
 axios
   .get('/api/sizes')
   .then((res) => {
     const view = new View(res.data)
 
-    const client = new Colyseus.Client(`ws://${window.location.host}${window.location.port ? ':' + window.location.port : ''}`)
-    const room = client.join('game')
+    const client = new Colyseus.Client(`ws://${window.location.hostname}${window.location.port ? ':' + window.location.port : ''}`)
+    client.joinOrCreate('game').then(room => {
+      const keymap = [
+        ['up', ['ArrowUp', 'w']],
+        ['down', ['ArrowDown', 's']],
+        ['left', ['ArrowLeft', 'a']],
+        ['right', ['ArrowRight', 'd']]
+      ]
 
-    const directions = ['up', 'down', 'left', 'right']
-    let virtuals = {}
-
-    directions.forEach((direction) => {
-      virtuals[direction] = {
-        keyup (transmitter) {
-          transmitter.send({ direction })
-        }
+      document.onkeydown = e => {
+        keymap.forEach(dir => {
+          if (dir[1].indexOf(e.key) + 1) return room.send('move', dir[0])
```

Comment:
`(dir[1].indexOf(e.key) + 1)` is just checking whether it's in the array right? If that's the case, how about:

```js
if (dir[1].indexOf(e.key) !== -1) return room.send('move', dir[0])

// or

if (dir[1].includes(e.key)) return room.send('move', dir[0])
```

Reply:
From my experience, `includes` has the worse performance, while `dir[1].indexOf(e.key) !== -1` looks less intuitive. But I have nothing against the last one.

Reply:
I think we'll keep it the way it is :+1: